src/components/Header/StatusBadge.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  Box,
+  HStack,
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  Text,
+} from "@chakra-ui/react"
+import { Badge } from "@opengovsg/design-system-react"
+import { BiCheckCircle, BiError, BiLoader } from "react-icons/bi"
+import { BsFillQuestionCircleFill } from "react-icons/bs"
+import { GoDotFill } from "react-icons/go"
+import { IconBaseProps } from "react-icons/lib"
+
+import { buildStatus } from "types/stagingBuildStatus"
+
+interface StatusBadgeProps {
+  status: buildStatus
+  timeLastSaved: number
+}
+
+interface StagingPopoverContentProps {
+  status: buildStatus
+  timeLastSavedInMin: number
+}
+
+const getTimeSavedInMins = (timeLastSaved: number): number => {
+  const timeLastSavedInMin = Math.floor(
+    (Date.now() - timeLastSaved) / 1000 / 60
+  )
+
+  return timeLastSavedInMin
+}
+
+const StagingPopoverContent = ({
+  status,
+  timeLastSavedInMin,
+}: StagingPopoverContentProps) => {
+  let headingText = ""
+  let bodyText = ""
+  let Icon = (props: IconBaseProps) => <BiLoader {...props} />
+  let timeText
+  if (timeLastSavedInMin < 60) {
+    timeText = `${timeLastSavedInMin} minute${
+      timeLastSavedInMin === 1 ? "" : "s"
+    } ago`
+  } else if (timeLastSavedInMin < 60 * 60) {
+    const hours = Math.floor(timeLastSavedInMin / 60)
+    timeText = `${hours} hour${hours === 1 ? "" : "s"} ago`
+  } else {
+    const days = Math.floor(timeLastSavedInMin / 60 / 24)
+    timeText = `${days} day${days === 1 ? "" : "s"} ago`
+  }
+  switch (status) {
+    case "READY":
+      headingText = `All saved edits from ${timeText} are on staging`
+      bodyText = "Click on 'Open staging' to take a look."
+
+      Icon = (props: IconBaseProps) => (
+        <BiCheckCircle {...props} color="#0F796F" />
```

Comment:
nit: can use `utility.feedback.success`?

Reply:
wait ah could you use this for icons? it didnt work for me idk why icons screws up

Reply:
I think can use Chakra's `<Icon>` then the `as` prop with the color tokens

Reply:
orh right ok uhm next monday release bah